ansible/inventory/multi_inventory.py

```diff
@@ -119,144 +227,103 @@ def load_yaml_config(self, conf_file=None):
-        if not env:
-            env = os.environ
-
-        # Allow relatively path'd providers in config file
-        if os.path.isfile(os.path.join(self.file_path, provider)):
-            provider = os.path.join(self.file_path, provider)
-
-        # check to see if provider exists
-        if not os.path.isfile(provider) or not os.access(provider, os.X_OK):
-            raise RuntimeError("Problem with the provider.  Please check path " \
-                        "and that it is executable. (%s)" % provider)
-
-        cmds = [provider]
-        if self.args.get('host', None):
-            cmds.append("--host")
-            cmds.append(self.args.get('host', None))
-        else:
-            cmds.append('--list')
-
-        if 'aws' in provider.lower():
-            cmds.append('--refresh-cache')
-
-        return subprocess.Popen(cmds, stderr=subprocess.PIPE, \
-                                stdout=subprocess.PIPE, env=env)
-
-    @staticmethod
-    def generate_config(provider_files):
-        """Generate the provider_files in a temporary directory.
-        """
-        prefix = 'multi_inventory.'
-        tmp_dir_path = tempfile.mkdtemp(prefix=prefix)
-        for provider_file in provider_files:
-            filedes = open(os.path.join(tmp_dir_path, provider_file['name']), 'w+')
-            content = Template(provider_file['contents']).substitute(tmpdir=tmp_dir_path)
-            filedes.write(content)
-            filedes.close()
-
-        return tmp_dir_path
-
-    def run_provider(self):
-        '''Setup the provider call with proper variables
-           and call self.get_provider_tags.
-        '''
+    def run_account_providers(self, force=False, valid_cache=True, account=None):
+        """Setup the provider call with proper variables and call self.get_provider_tags"""
         try:
-            all_results = []
-            tmp_dir_paths = []
+            # if account parameter is present we only want refresh that specific account
+            tmp_accounts = self.accounts
+            if account:
+                tmp_accounts = [acc for acc in self.accounts if acc.name == account]
+
             processes = {}
-            for acc_name, account in self.config['accounts'].items():
-                tmp_dir = None
-                if account.has_key('provider_files'):
-                    tmp_dir = MultiInventory.generate_config(account['provider_files'])
-                    tmp_dir_paths.append(tmp_dir)
-
-                # Update env vars after creating provider_config_files
-                # so that we can grab the tmp_dir if it exists
-                env = account.get('env_vars', {})
-                if env and tmp_dir:
-                    for key, value in env.items():
-                        env[key] = Template(value).substitute(tmpdir=tmp_dir)
-
-                provider = account['provider']
-                processes[acc_name] = self.get_provider_tags(provider, env)
+            for acc in tmp_accounts:
```

Comment:
If there are multiple accounts with the same name, shouldn't we just error instead of looping?